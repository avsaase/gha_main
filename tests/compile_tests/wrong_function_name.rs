use gha_main::{gha_main, output, GitHubActionResult};

#[gha_main]
fn not_main() -> GitHubActionResult {
    let value = "value";
    Ok(output!(value))
}
