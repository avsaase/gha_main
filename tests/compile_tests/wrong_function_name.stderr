error: function must be called `main`
 --> tests/compile_tests/wrong_function_name.rs:4:4
  |
4 | fn not_main() -> GitHubActionResult {
  |    ^^^^^^^^

error[E0432]: unresolved import `gha_main::output`
 --> tests/compile_tests/wrong_function_name.rs:1:26
  |
1 | use gha_main::{gha_main, output, GitHubActionResult};
  |                          ^^^^^^ no `output` in the root

warning: unused import: `GitHubActionResult`
 --> tests/compile_tests/wrong_function_name.rs:1:34
  |
1 | use gha_main::{gha_main, output, GitHubActionResult};
  |                                  ^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/compile_tests/wrong_function_name.rs:7:2
  |
7 | }
  |  ^ consider adding a `main` function to `$DIR/tests/compile_tests/wrong_function_name.rs`
